src/cpp/rtps/transport/TCPTransportInterface.cpp

```diff
@@ -972,6 +980,11 @@ bool TCPTransportInterface::Receive(
             close_tcp_socket(channel);
             success = false;
         }
+        else if (!channel->connection_status())
+        {
+            logWarning(DEBUG, "Error reading TCP header: channel disconnected while reading.");
```

Comment:
Should the log category be different?

Reply:
Currently there are no too many log levels supported. If you don't want to promote this log to the Error level (even though the message says "Error") then a Warning seems to be correct. If the Warning level is kept, I suggest changing the log message:

```suggestion
            logWarning(DEBUG, "Failed to read TCP header: channel disconnected while reading.");
```

Reply:
Yes I believe it should be a warning, though I was referring to the `DEBUG` part.

Reply:
Other Errors/Warnings in this module are being registered under the `DEBUG` label. Thus I will keep consistency. Either change all other logs to another new label (maybe `TCP_DEBUG`) or keep this label as it is currently.